contracts/bridge/BridgeOperator.sol

```diff
@@ -51,29 +56,40 @@ contract BridgeOperator is Ownable {
     }
 
     // voteCommon handles common functionality for voting.
-    function voteCommon(VoteType voteType, bytes32 _voteKey)
+    function voteCommon(VoteType _voteType, uint64 _nonce, bytes32 _voteKey)
         internal
         returns(bool)
     {
-        if (!votes[_voteKey][msg.sender]) {
-            votes[_voteKey][msg.sender] = true;
-            require(votesCounts[_voteKey] < votesCounts[_voteKey] + 1, "votesCounts overflow");
-            votesCounts[_voteKey]++;
+        VotesData storage vote = votes[uint8(_voteType)][_nonce];
+
+        if (!vote.voted[_voteKey][msg.sender]) {
+            vote.voted[_voteKey][msg.sender] = true;
+
+            require(vote.voteCounts[_voteKey] < vote.voteCounts[_voteKey] + 1, "voteCounts overflow");
```

Comment:
operatorThresholds is uint8. Is this condition really possible?

Reply:
Right I removed it. :)